Sources/Classes/RSMoEngageDestination.swift

```diff
@@ -6,62 +6,83 @@
             client?.log(message: "Failed to Initialize MoEngage Factory", logLevel: .warning)
             return
         }
+        
+        // Redirect data according to region
+        if moEngageConfig.region == "EU" {
+            moEngageDataCenter = .data_center_02
+        } else if moEngageConfig.region == "US" {
+            moEngageDataCenter = .data_center_01
+        } else if moEngageConfig.region == "IND" {
+            moEngageDataCenter = .data_center_03
+        }
+        else{
+            client?.log(message:"MoEngage SDK initialization terminated due to an invalid region.", logLevel: .warning)
+        }
+        
+        let sdkConfig = MoEngageSDKConfig(appId: moEngageConfig.apiId, dataCenter: moEngageDataCenter!)
         // Check if debug mode is on or off
 #if DEBUG
-        MoEngage.sharedInstance().initializeDev(withAppID: moEngageConfig.apiId)
+      
+        MoEngage.sharedInstance.initializeDefaultTestInstance(sdkConfig)
 #else
-        MoEngage.sharedInstance().initializeProd(withAppID: moEngageConfig.apiId)
-#endif
-        if moEngageConfig.region == "EU" {
-            MoEngage.redirectData(to: MOE_REGION_EU)
-        }
         
+        MoEngage.sharedInstance.initializeDefaultLiveInstance(sdkConfig)
+#endif
+       
         if UNUserNotificationCenter.current().delegate == nil {
            UNUserNotificationCenter.current().delegate = self
         }
         
         client?.log(message: "Initializing MoEngage SDK", logLevel: .debug)
     }
-
+    
     func identify(message: IdentifyMessage) -> IdentifyMessage? {
         reset()
         if let anonymousId = message.anonymousId {
-            MoEngage.sharedInstance().setUserAttribute(anonymousId, forKey: "anonymousId")
+          
+            MoEngageSDKAnalytics.sharedInstance.setUserAttribute(anonymousId, withAttributeName: "anonymousId")
         }
+        
         if let userId = message.userId {
-            MoEngage.sharedInstance().setUserAttribute(userId, forKey: USER_ATTRIBUTE_UNIQUE_ID)
-            MoEngage.sharedInstance().setUserUniqueID(userId)
+            MoEngageSDKAnalytics.sharedInstance.setUserAttribute(userId, withAttributeName: "userId")
+            MoEngageSDKAnalytics.sharedInstance.setUniqueID(userId)
         }
         
         if let traits = message.traits, !traits.isEmpty {
             handle(traits: traits)
+            
         }
         return message
     }
+    
+    
+
 
     func track(message: TrackMessage) -> TrackMessage? {
         if !message.event.isEmpty {
             switch message.event {
-            case RSEvents.LifeCycle.applicationInstalled: MoEngage.sharedInstance().appStatus(INSTALL)
-            case RSEvents.LifeCycle.applicationUpdated: MoEngage.sharedInstance().appStatus(UPDATE)
+            case RSEvents.LifeCycle.applicationInstalled: MoEngageAppStatus(rawValue: 0)
+            case RSEvents.LifeCycle.applicationUpdated: MoEngageAppStatus.update
```

Comment:
In this code, you're not setting the app status. `MoEngageAppStatus.update` doesn't seem to do anything.

Please find the relevant code snippet from the MoEngage doc.